src/browser/core/servers.ts

```diff
@@ -41,10 +49,10 @@ export async function update(server: Server, cryptoSecret: string): Promise<Serv
 
   await config.save(data);
 
-  return server;
+  return { data: srv };
```

Comment:
While working on this ticket, I noticed a bug in this code, we were returning the wrong object. It should be returning the saved record, which has the encrypted password, just like we do for the `add` function, but it was returning the incoming data with the decrypted password.